contracts/contracts/strategies/AerodromeEthStrategy.sol

```diff
@@ -0,0 +1,592 @@
+        address _recipient,
+        address _weth,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(_amount > 0, "Invalid amount");
+        require(_weth == address(weth), "Can only withdraw WETH");
+
+        emit Withdrawal(_weth, address(lpTokenAddress), _amount);
+
+        uint256 requiredLpTokens = calcTokenToBurn(_amount);
+
+        _lpWithdraw(requiredLpTokens);
+
+        /* math in requiredLpTokens should correctly calculate the amount of LP to remove
+         * in that the strategy receives enough WETH on balanced removal
+         */
+        // slither-disable-next-line unused-return
+        aeroRouterAddress.removeLiquidity(
+            address(weth),
+            address(oeth),
+            true,
+            requiredLpTokens,
+            _amount,
+            0,
+            address(this),
+            block.timestamp
+        );
+
+        // Burn all the removed OETH and any that was left in the strategy
+        uint256 oethToBurn = oeth.balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(oethToBurn);
+
+        emit Withdrawal(address(oeth), address(lpTokenAddress), oethToBurn);
+
+        require(
+            weth.transfer(_recipient, _amount),
+            "Transfer of WETH not successful"
+        );
+    }
+
+    function calcTokenToBurn(uint256 _wethAmount)
+        internal
+        view
+        returns (uint256 lpToBurn)
+    {
+        /* The rate between coins in the pool determines the rate at which pool returns
+         * tokens when doing balanced removal (removeLiquidity call). And by knowing how much WETH
+         * we want we can determine how much of OETH we receive by removing liquidity.
+         *
+         * Because we are doing balanced removal we should be making profit when removing liquidity in a
+         * pool tilted to either side.
+         *
+         * Important: A downside is that the Strategist / Governor needs to be
+         * cognisant of not removing too much liquidity. And while the proposal to remove liquidity
+         * is being voted on the pool tilt might change so much that the proposal that has been valid while
+         * created is no longer valid.
+         */
+
+        uint256 poolWETHBalance = wethCoinIndex == 0
+            ? lpTokenAddress.reserve0()
+            : lpTokenAddress.reserve1();
+
+        /* K is multiplied by 1e36 which is used for higher precision calculation of required
+         * pool LP tokens. Without it the end value can have rounding errors up to precision of
+         * 10 digits. This way we move the decimal point by 36 places when doing the calculation
+         * and again by 36 places when we are done with it.
+         */
+        uint256 k = (1e36 * IERC20(address(lpTokenAddress)).totalSupply()) /
+            poolWETHBalance;
+        // prettier-ignore
+        // slither-disable-next-line divide-before-multiply
+        uint256 diff = (_wethAmount + 1) * k;
```

Comment:
This calculation currently calculates a ratio, then uses that ratio to calculate the end result. This is slightly lossy in doing two calculations with rounding errors, which requires the extra high intermediate precision.

Would this be much simpler and less  if we just move the (_wethAmount+1) to be directly multiplied by the totalSupply in the first step?

Something like:

`(_wethAmount+1) * totalSupply  / poolWETHBalance` ?

Reply:
I tired using multiple values, it doesn't make any difference. so yea we can use the one you suggested, might save some gas too. 

| Action   | Value                         | Formula 1               | Formula 2               |
|----------|-------------------------------|-------------------------|-------------------------|
| withdraw | 220                           | 82862338377910162699    | 82862338377910162699    |
| withdraw | 1                             | 1035779229723877034     | 1035779229723877034     |
| withdraw | 0.0000000001                  | 103577924               | 103577924               |
| withdrawAll | 30000000000000000001       | 30000000000000000001    | 30000000000000000001    |